src/Microsoft.DotNet.Build.Tasks/GenerateBindingRedirect.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.IO;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateBindingRedirect : Task
+    {
+        [Required]
+        public ITaskItem[] AssemblyIdentities { get; set; }
+
+        [Required]
+        public ITaskItem[] Executables { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        private static XNamespace ns { get; set; }
+        public override bool Execute()
+        {
+            ns = "urn:schemas-microsoft-com:asm.v1";
+            XElement bindingRedirectAssemblies = new XElement(ns + "assemblyBinding");
+            foreach (ITaskItem assembly in AssemblyIdentities)
+            {
+                string assemblyVersion = assembly.GetMetadata("Version");
+                XElement assemblyIdentity = new XElement("assemblyIdentity",
+                                                         new XAttribute("name", assembly.GetMetadata("Name")),
+                                                         new XAttribute("publicKeyToken", assembly.GetMetadata("PublicKeyToken")),
```

Comment:
why not do the same for PublicKeyToken and provide a default if we don't have one?